osf/migrations/__init__.py

```diff
@@ -135,4 +152,33 @@ def update_waffle_flags(sender, verbosity=0, **kwargs):
 
 def create_cache_table(sender, verbosity=0, **kwargs):
     if getattr(sender, 'label', None) == 'osf':
-        call_command('createcachetable', tablename=settings.CACHES[settings.STORAGE_USAGE_CACHE_NAME]['LOCATION'])
+        call_command('createcachetable', tablename=api_settings.CACHES[api_settings.STORAGE_USAGE_CACHE_NAME]['LOCATION'])
+
+
+def update_default_providers(sender, verbosity=0, **kwargs):
+    if getattr(sender, 'label', None) == 'osf':
+        if 'pytest' in sys.modules:
+            ensure_default_registration_provider()
+
+
+def ensure_default_providers():
+    ensure_default_preprint_provider()
+    ensure_default_registration_provider()
+
+
+def ensure_default_preprint_provider():
+    PreprintProvider = apps.get_model('osf', 'PreprintProvider')
+
+    PreprintProvider.objects.update_or_create(
+        _id=OSF_PREPRINTS_PROVIDER_DATA['_id'],
```

Comment:
One related concern: if we change any of the following provider info in admin, does running this as post-migration overwrite the admin changes.

```
OSF_PREPRINTS_PROVIDER_DATA = {
    '_id': 'osf',
    'name': 'Open Science Framework',
    'domain': settings.DOMAIN,
    'share_publish_type': 'Preprint',
    'domain_redirect_enabled': False,
}

OSF_REGISTRIES_PROVIDER_DATA = {
    '_id': 'osf',
    'name': 'OSF Registries',
    'domain': settings.DOMAIN,
    'share_publish_type': 'Registration',
    'domain_redirect_enabled': False,
}
```

(I admit the chance of these fields being changed is very low)

Reply:
I took my suggestion back in https://github.com/CenterForOpenScience/osf.io/pull/9961#discussion_r919237826 since defining license may mess up with existing licenses for production `osf` providers.

One side effect of removing licenses from provider creation:  if provider is created (which happens during unit tests or in clean local environment), we need to add license explicitly before using the provider in the code for unit tests and via admin for running local OSF.

My question is: can we have a provider with license being undefined?